src/azstoragetorch/_client.py

```diff
@@ -185,17 +219,32 @@ def _download_with_retries(self, pos: int, length: int) -> bytes:
 
     def _get_download_stream(self, pos: int, length: int) -> Iterator[bytes]:
         try:
-            return self._generated_sdk_storage_client.blob.download(
+            response = self._generated_sdk_storage_client.blob.download(
                 range=f"bytes={pos}-{pos + length - 1}",
-                modified_access_conditions=azure.storage.blob._generated.models.ModifiedAccessConditions(
+            ) 
+            if self._blob_size is None: 
+                headers = response.response.headers
+                if "Content-Range" not in headers:
+                    raise ValueError("Content-Range header not found in response headers")
```

Comment:
I don't think we need to explicitly raise if the header is not present. We always expect it to be present and so we should go ahead with retrieving the header name and if it is not present there will be a propagated on our behalf from the dictionary anyway.

In general, we should avoid raising validation errors on responses as if we are not careful, the validation can backfire and cause the library to throw errors when we were getting successful, valid responses.